util/aso/defaults.go

```diff
@@ -0,0 +1,24 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aso
+
+import "fmt"
+
+// GetASOSecretName formats the name of the ASO Secret created by the capz controller.
+func GetASOSecretName(owner, subscriptionID string) string {
+	return fmt.Sprintf("%s-%s", owner, subscriptionID)
```

Comment:
is subscriptionID unique enough? Do we expect to only ever have 1 secret per subscription + owner (I'm assuming owner is the cluster here)?

Reply:
I would say it should be, with the caveat mentioned in https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3698#discussion_r1269407349 . Owner can be a cluster if it doesn't specify an AzureClusterIdentity ref or an AzureClusterIdentity if the cluster has defined one.